code/modules/clothing/glasses/tajblind.dm

```diff
@@ -0,0 +1,196 @@
+	icon_state = "tajblind"
+	item_state = "tajblind"
+	actions_types = list(/datum/action/item_action/toggle)
+	color_view = MATRIX_STANDARD
+	correct_wires = TRUE
+	var/list/modes = list(MODE_OFF = MODE_NATURAL, MODE_NATURAL = MODE_CORRECTION, MODE_CORRECTION = MODE_OFF)
+	var/selected_mode = MODE_CORRECTION
+
+	sprite_sheets = list(
+		"Vox" = 'icons/mob/clothing/species/vox/eyes.dmi',
+		"Grey" = 'icons/mob/clothing/species/grey/eyes.dmi'
+		)
+
+/obj/item/clothing/glasses/hud/tajblind/attack_self(mob/user)
+	toggle_veil(user, TRUE)
+
+/obj/item/clothing/glasses/hud/tajblind/proc/toggle_veil(mob/user, voluntary)
+	var/mob/living/carbon/human/H = user
+	selected_mode = modes[selected_mode]
+	to_chat(user, "<span class='[voluntary ? "notice" : "warning"]'>[voluntary ? "You turn the veil" : "The veil turns"] [selected_mode ? "to [selected_mode] mode" : "off"][voluntary ? "." : "!"]</span>")
+
+	switch(selected_mode)
+		if(MODE_OFF)
+			tint = TINT_BLIND
+			flash_protect = FLASH_PROTECTION_WELDER
+			color_view = null
+			correct_wires = FALSE
+
+		if(MODE_NATURAL)
+			tint = initial(tint)
+			flash_protect = initial(flash_protect)
+			color_view = null
+
+		if(MODE_CORRECTION)
+			tint = initial(tint)
+			flash_protect = initial(flash_protect)
+			color_view = MATRIX_STANDARD
+			correct_wires = TRUE
+
+	if(ishuman(H))
+		if(H.glasses == src)
+			H.update_sight()
+			H.update_client_colour()
+
+/obj/item/clothing/glasses/hud/tajblind/examine_more(mob/user)
+	. = ..()
+	. += "Tajaran Veils have long been an important part of their spirituality and culture, suppressed by the Overseers and making a strong return after the civil war. Tajaran believe that to see one’s eyes is to see their soul, and thus the more spiritual Tajara wear veils to conceal their eyes from everyone but the ones closest to them. <br>\
+			These current designs are adapted from recreations of the ancient veils, created by the Alchemists Guild. Technologically advanced and created to help Tajara adapt to life in the larger galactic community, they have systems built-in that allow them to have holographic huds, as well as corrective technology to help Tajaran overcome their genetic tritanopia colour blindness. <br>\
+			Availability on the wider market is highly restricted as a result of their cultural importance, as well as the patent held by the Alchemists Guild, and the lenses are very hard to reverse engineer. Popular theories suggest this as a result of the unique materials available on Adhomai, or the inability to recreate the light conditions of the Tajara homeworld."
+
+/obj/item/clothing/glasses/hud/tajblind/meson
+	name = "\improper Tajaran engineering meson veil"
+	icon_state = "tajblind_engi"
+	item_state = "tajblind_engi"
+
+/obj/item/clothing/glasses/hud/tajblind/meson/Initialize()
+	..()
+	desc += "<br><span class='info'>It has an optical meson scanner integrated into it.</span>"
```

Comment:
Move this out into `examine()`.

Reply:
The problem with this and everything below is that glasses have `<span class='info'>You can <b>Alt-Click</b> [src] to adjust if it fits over or under your mask.</span>` attached to the end of their `examine()`.
If I move it there, the Alt-Click line will end up not being the last line of the examine. So while it'd be more consistent code-wise, the user-facing part would become inconsistent with other glasses.

Reply:
That's fair. You can probably leave this as-is for now, then.